### ISSUE TITLE
Various issues with retroachievements on PCSX Rearmed

### ISSUE THREAD

**Issue by brunocar** [spruceUI/spruceOS#33]
Initially retroarchievements seemed to specifically not work on PCSX, using retroarch expert mode to access the retroachievements menu, the same set up on every other core worked fine.

Upon further inspection, manually booting the core from retroarch seemed to suddenly do the trick, but that still presented 2 problems: M3U playlists (used for multidisk games, which PSX has quite a bit of) still arent recognized by retroachievements (works fine on PC retroarch and rocknix, for reference) and opening the retroachievements menu seems to not fully load all the achievements and icons, and quickly afterwards crash retroarch, interestingly, leaving the game alone for a few minutes without opening the retroarch menu fixes it, seemingly permanently for that game (IE: its likely that the achievements are precached in the background and the wifi is too slow for what retroarch expects for this feature) but this has to be done for every single new game and seems to be a PSX exclusive problem (havent tested if duckstation has this same problem).

Right now the only problem i havent figured out a solution to is the multidisk one, i got parasite eve 1 to detect it but for some reason every other one just refuses to do so, despite retroarch reporting the correct hash.

**Comment by brunocar** [comment]
Further testing proved 2 things: the multidisk issue wasnt what i thought it was, basically it seems all games show the same problem of requiring that you open them a couple of times till it does the network check BEFORE starting the game, which retroachievements needs, thats one problem, but the other is that yes, consistently, it seems that even if it works (and the correct notif shows up) it takes a long while for it to precache the achievements and retroarch straight up crashes if you try to see them before its finished.
the first problem might get fixed with some sort of "wait for network" option that retroarch might have, but the second one seems to be simply that retroarch doesnt expect internet connections this slow for this task and doesnt know what to do with it, if so we might have to report that to them.

**Comment by brunocar** [comment]
I've talked to tGecko from the onionOS team and they made their own solution to the first problem as retroarch didnt have one, to quote him:
There is a setting to wait for time sync at startup, which includes waiting for network, find it in 
Tweaks -> System -> Date and time -> Wait for sync on startup
You need to enable time syncing for it to be available

It might be worth looking into those scripts and adding them to spruce.

**Comment by brunocar** [comment]
closed in order to make a new issue about the true root of the problem
